netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -198,7 +213,27 @@ public Runnable start(Listener transportListener) {
      * that it may begin buffering writes.
      */
     b.handler(negotiationHandler);
-    channel = b.register().channel();
+    ChannelFuture regFuture = b.register();
```

Comment:
This is discussed some in the comment below that talks about "we are calling lifecycleManager from the event loop". Basically, I need `statusExplainingWhyTheChannelIsNull` to be computed before `start()` returns, and I feel like adding the listener within the `if (channel == null)` doesn't help much vs what I've done.